checkstyle/suppressions.xml

```diff
@@ -161,7 +161,7 @@
               files="StreamThread.java"/>
 
     <suppress checks="ClassDataAbstractionCoupling"
-              files="(KStreamImpl|KTableImpl).java"/>
+              files="(KafkaStreams|KStreamImpl|KTableImpl).java"/>
```

Comment:
Is this needed due to the `AtomicInteger`?

Reply:
I believe it is.

Since `AtomicInteger` is more of a "data type" than "another class", I'd be fine with just not counting it toward the check:

```
 <module name="ClassDataAbstractionCoupling">
   <property name="excludeClassesRegexps" value="AtomicInteger"/>
 </module>
```

cf: https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.html

Reply:
Thanks for the proposal! I did not know about that possibility. 
Every time, I see changes to this file an alarm goes off in my head. I think you are right we need to discuss how much we value the code style rules. In general, I think they are a good sign that we need to refactor our code and I am a bit afraid that we will never do it, if we keep adding suppression rules. In this specific case, I am fine with excluding `AtomicInteger` from the check. At least, we get the error if we add too many references to other objects. Maybe, we should open a newbie ticket each time we think that we add a suppression.